### ISSUE TITLE
WCS results from /evaluate not preprocessed in /converse

### ISSUE THREAD

**Issue by M0rt1mer** [Watson-Personal-Assistant/skill-sdk-nodejs#15]
When "skill" is used as NLU, WA core calls /evaluate on skill, which in turn probably calls WCS, and the result of WCS is sent to WA core as context (this is all handled by **handler**). Then core calls /converse, at which point an action is selected in handler. This action can access WCS context inside request.evaluationResponse.context, however this context is in different format than regular context, which is extremely inconvenient to work with. You basically need to replicate **createContext** function from handler.
Common use case, I imagine, is trying to send the evaluationResponse context in response to /converse. That way it is saved as **skill context** in core, and then used in evaluation in next user interaction - thus allowing multi-step dialogs.
tl;dr
Handler should run **request.evaluationResponse** through **createContext** before passing it to appropriate action handler.

**Comment by offerakrabi** [comment]
you can use the saveEvaluationContext function from the handler:
its parameters are the context variable and the context from the evaluation
(handler.saveEvaluationContext(context, evaluationContext))

**Comment by M0rt1mer** [comment]
I didn't notice that, thanks. Works fine
